Traduction of the "Managing state" page (#458)
reactjs/fr.react.dev: src/content/learn/managing-state.md

</Sandpack>

> Vous pouvez le retirer et simplifier le code en évaluant `fullName` pendant que le composant s'affiche :

Reviewer: ```suggestion
Vous pouvez le retirer et simplifier le code en calculant `fullName` à l’affichage du composant :
```

  return (
    <>

>       <h2>Vérifions votre identité :</h2>

Reviewer: ```suggestion
      <h2>Enregistrez-vous :</h2>
```

      <label>
        Prénom :{' '}
        <input
          value={firstName}
          onChange={handleFirstNameChange}
        />
      </label>
      <label>
        Nom :{' '}
        <input
          value={lastName}
          onChange={handleLastNameChange}
        />
      </label>
      <p>

>         Votre billet sera délivré à : <b>{fullName}</b>

Reviewer: ```suggestion
        Votre billet sera au nom de : <b>{fullName}</b>
```

</Sandpack>

> Cela peut sembler être un petit changement, mais de nombreux bugs dans les applications React sont corrigés de cette façon.

Reviewer: ```suggestion
Ce changement peut sembler mineur, mais de nombreux bugs dans les applications React se corrigent de cette façon.
```

<LearnMore path="/learn/choosing-the-state-structure">

Lisez **[Choisir la structure de l'état](/learn/choosing-the-state-structure)** pour apprendre à concevoir la forme de l'état afin d'éviter les bugs.

</LearnMore>

> ## Partager l'état entre les composants {/*sharing-state-between-components*/}

Reviewer: ```suggestion
## Partager l'état entre des composants {/*sharing-state-between-components*/}
```

<LearnMore path="/learn/preserving-and-resetting-state">

> Lisez **[Préserver et réinitialiser l'état](/learn/preserving-and-resetting-state)** pour apprendre la durée de vie d'un état et comment la contrôler.

Reviewer: ```suggestion
Lisez **[Préserver et réinitialiser l'état](/learn/preserving-and-resetting-state)** pour apprendre le cycle de vie d'un état et comment le contrôler.
```

Erreur quasi-certaine dans la VO : "lifetime" au lieu de "lifecycle".

</LearnMore>

> ## Extraire la logique d'état avec un réducteur {/*extracting-state-logic-into-a-reducer*/}

Reviewer: ```suggestion
## Extraire la logique d'état dans un réducteur {/*extracting-state-logic-into-a-reducer*/}
```

> Les composants comportant de nombreuses mises à jour d'état réparties entre de nombreux gestionnaires d'événements peuvent devenir encombrants. Dans ce cas, vous pouvez consolider toute la logique de mise à jour de l'état en dehors de votre composant dans une seule fonction, appelée "reducer". Vos gestionnaires d'événements deviennent concis car ils ne spécifient que les "actions" de l'utilisateur. Au bas du fichier, la fonction reducer spécifie comment l'état doit être mis à jour en réponse à chaque action !

Reviewer: ```suggestion
Les composants comportant de nombreuses mises à jour d'état réparties entre de nombreux gestionnaires d'événements peuvent devenir intimidants. Dans ce type de cas, vous pouvez consolider toute la logique de mise à jour de l'état en-dehors de votre composant dans une seule fonction, appelée « réducteur » _(reducer, NdT)_. Vos gestionnaires d'événements deviennent concis car ils ne spécifient que les « actions » de l'utilisateur. Au bas du fichier, la fonction du réducteur spécifie comment l'état doit être mis à jour en réponse à chaque action !
```

Reviewer: Très propre !

  return (
    <>
      <input

>         placeholder="Ajouter tâche"

Reviewer: ```suggestion
        placeholder="Ajouter une tâche"
```

      <>
        {task.text}
        <button onClick={() => setIsEditing(true)}>

>           Editer

Reviewer: ```suggestion
          Modifier
```

<LearnMore path="/learn/extracting-state-logic-into-a-reducer">

> Lisez **[Extraire la logique d'état avec un réducteur](/learn/extracting-state-logic-into-a-reducer)** pour apprendre à consolider la logique dans la fonction "reducer".

Reviewer: ```suggestion
Lisez **[Extraire la logique d'état dans un réducteur](/learn/extracting-state-logic-into-a-reducer)** pour apprendre à consolider la logique au sein d’une fonction réducteur.
```

Une trad trop "fidèle" est ampoulée…

</LearnMore>

## Transmettre des données en profondeur avec le contexte {/*passing-data-deeply-with-context*/}

Habituellement, vous transmettez des informations d'un composant parent à un composant enfant par l'intermédiaire des props. Mais le passage des props peut s'avérer gênant si vous devez faire passer une information à travers plusieurs composants, ou si plusieurs composants ont besoin de la même information. 

> Le contexte permet au composant parent de rendre certaines informations disponibles à n'importe quel composant de l'arbre situé en dessous de lui, quelle que soit sa profondeur, sans avoir à les transmettre explicitement par le biais de props.

Reviewer: ```suggestion
Le contexte permet au composant parent de rendre certaines informations disponibles à n'importe quel composant de l'arbre situé en-dessous de lui--quelle que soit sa profondeur--sans avoir à les transmettre explicitement par le biais de props.
```

Nickel 👏

> Ici, le composant `Heading` détermine son niveau d'en-tête en "demandant" son niveau à la `Section` la plus proche. Chaque `Section` suit son propre niveau en demandant à la `Section` parente et en lui en ajoutant un. Chaque `Section` fournit des informations à tous les composants situés en dessous d'elle sans passer de props - elle le fait par le biais du contexte.

Reviewer: ```suggestion
Ici, le composant `Heading` détermine son niveau d'en-tête en "demandant" son niveau à la `Section` la plus proche. Chaque `Section` détermine son propre niveau en demandant à la `Section` parente le sien, et en lui ajoutant un. Chaque `Section` fournit des informations à tous les composants situés en-dessous d'elle sans passer de props--elle le fait par le biais du contexte.
```

    case 6:
      return <h6>{children}</h6>;
    default:

>       throw Error('Niveau inconnu: ' + level);

Reviewer: ```suggestion
      throw Error('Niveau inconnu : ' + level);
```

  return (
    <>
      <input

>         placeholder="Ajouter tâche"

Reviewer: ```suggestion
        placeholder="Ajouter une tâche"
```

      <>
        {task.text}
        <button onClick={() => setIsEditing(true)}>

>           Editer

Reviewer: ```suggestion
          Modifier
```

<LearnMore path="/learn/scaling-up-with-reducer-and-context">

> Lisez **[Mise à l'échelle avec un réducteur et un contexte](/learn/scaling-up-with-reducer-and-context)** pour apprendre comment la gestion des états s'adapte à une application en pleine croissance.

Reviewer: ```suggestion
Lisez **[Mise à l'échelle avec réducteur et contexte](/learn/scaling-up-with-reducer-and-context)** pour apprendre comment adapter la gestion d’état à une application en pleine croissance.
```

</LearnMore>

> ## Quelle est la suite ? {/*whats-next*/}

Reviewer: ```suggestion
## Et maintenant ? {/*whats-next*/}
```

Traduction entérinée.

> Rendez vous sur [Réagir à une entrée avec un état](/learn/reacting-to-input-with-state) pour commencer à lire ce chapitre page par page !

Reviewer: ```suggestion
Allez sur [Réagir à la saisie avec un état](/learn/reacting-to-input-with-state) pour commencer à lire ce chapitre page par page !
```

> Ou, si vous êtes déjà familier avec ces sujets, pourquoi ne pas lire à propos des [Trappes d'évacuation](/learn/escape-hatches)?

Reviewer: ```suggestion
Ou alors, si vous êtes déjà à l’aise avec ces sujets, pourquoi ne pas explorer les [Échappatoires](/learn/escape-hatches)?
```

Note que le terme VO m'embête à la base, j'en parle avec Dan. Mettre tout ce qui touche aux effets et hooks personnalisés dans "Escape hatches", même si je comprends l'intention, me semble problématique en termes de discours.

La trad entérinée est en tout cas "échappatoires".
